src/DotLiquid/Tags/For.cs

```diff
@@ -223,18 +221,43 @@ private static List<object> SliceCollectionUsingEach(Context context, IEnumerabl
             }
             return segments;
         }
+    }
+
+    /// <summary>
+    /// internal class to encapsulate pre DotLiquid 2.2 compatibility in for loop.
+    /// </summary>
+    /// <remarks>
+    /// DotLiquid 2.2 compatibility includes:
+    /// * An undocumented property `itemName` which is an implicit alias for KeyValuePair.Key
+    /// * Implicit access of properties within a nested IDictionary Value
+    /// </remarks>
+    class LegacyKeyValueDrop : Drop

```

Comment:
should this be marked internal?

Reply:
Isn't internal the default visibility for a class?

Reply:
Better make it explicit that leave the reader to guess.